WordPress/Sniffs/Theme/RestrictedPHPFunctionsSniff.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Forbids usage of certain fuctions and recommends alternatives.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.xx.0
+ */
+class WordPress_Sniffs_Theme_RestrictedPHPFunctionsSniff extends WordPress_Sniffs_Functions_FunctionRestrictionsSniff {
+
+	/**
+	 * Groups of functions to restrict
+	 *
+	 * Example: groups => array(
+	 * 	'lambda' => array(
+	 * 		'type'      => 'error' | 'warning',
+	 * 		'message'   => 'Use anonymous functions instead please!',
+	 * 		'functions' => array( 'eval', 'create_function' ),
+	 * 	)
+	 * )
+	 *
+	 * @return array
+	 */
+	public function getGroups() {
+		return array(
+
+			'eval' => array(
+				'type'      => 'error',
+				'message'   => '%s() is not allowed.',
+				'functions' => array(
+					'eval',
+				),
+			),
+
+			'system_calls' => array(
```

Comment:
This group should be removed as it's already covered in the `PHP\DiscouragedPHPFunctions` sniff.
If the error level & message needs changing, this can be done from the ruleset.